docs/source/api/user-guide/combinators/compositors.rst

```diff
@@ -0,0 +1,387 @@
+the privacy properties of this compositor hold in the interactive
+setting, the compositor must lock, or freeze, any queryable it has
+previously spawned when a new query arrives.
+
+This is an artifact of how non-interactive composition results have been
+extended to work in the interactive setting. Namely, that the second
+query can be viewed as a postprocessing of the first query.
+Unfortunately, this postprocessing argument doesn’t necessarily hold
+when the analyst may still interact with the first queryable. This is
+the subject of a further line of research on concurrent compositors,
+which we hope to make available in the next library release.
+
+An example of this constraint is demonstrated in the “Nesting” section
+below.
+
+Chaining
+--------
+
+Since all compositors are just “plain-old-measurements” they also
+support chaining.
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> str_space = dp.vector_domain(dp.atom_domain(T=str)), dp.symmetric_distance()
+            >>> str_sc_meas = str_space >> dp.t.then_cast_default(int) >> sc_meas
+            
+            >>> str_sc_qbl = str_sc_meas(["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"])
+            >>> str_sc_qbl(sum_meas), str_sc_qbl(count_meas)
+            (..., ...)
+
+``str_sc_meas`` is invoked with a string dataset, but returns a
+queryable that takes queries over integer datasets. Chaining compositors
+can be used to avoid repeating the same transformations for each query.
+
+Keep in mind that the ``d_in`` on the interactive compositor must match
+the output distance from the previous transformation:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> max_contributions = 1
+            >>> sum_trans = input_space >> dp.t.then_clamp((0, 10)) >> dp.t.then_sum()
+            >>> sc_meas = sum_trans >> dp.c.make_sequential_composition(
+            ...     input_domain=sum_trans.output_domain,
+            ...     input_metric=sum_trans.output_metric,
+            ...     output_measure=dp.max_divergence(T=float),
+            ...     d_in=sum_trans.map(max_contributions),
+            ...     d_mids=[2., 1.]
+            ... )
+            
+
+In this code snip, we used the supporting elements and map from the
+transformation to fill in arguments to the sequential compositor
+constructor, and to derive a suitable ``d_in`` for the compositor, based
+on a known ``d_in`` for the sum transformation.
+
+Nesting
+-------
+
+Just like in chaining, since all compositors are
+“plain-old-measurements” they can also be used as arguments to
+interactive compositors. In this example, we nest a zCDP sequential
+compositor inside an approximate-DP sequential compositor.
+
+We first make the approximate-DP sequential compositor, accepting two
+queries. The first query must be :math:`(2 ε, 10^{-6} δ)`-DP, and the
+second :math:`(1 ε, 0 δ)`-DP.
```

Comment:
We should not use ε, δ as units. Commenting to open in new issue.

Reply:
- filed #1805